File: lista3/zadanie2/src/Rectangle.cpp
#include "Rectangle.hpp"

bool Rectangle::CheckIfItCouldBe(ClientInput input)
{
    std::set<double> checkSides = readSidesFromInput(input);

    if(checkSides.size() > 2)
        return false;
    if(input.NextInput() != 90.)
        return false;
    return true;
}

Rectangle::Rectangle(ClientInput input) : Quadrangle(input)
{}

double Rectangle::Area()
{
    std::set<double>::iterator iterator = sides.begin();
    return (*iterator) * (*++iterator);
}

double Rectangle::Perimeter()
{
    std::set<double>::iterator iterator = sides.begin();
    return 2.*(*iterator)  + 2.*(*++iterator);
}

std::string Rectangle::NameYourself()
{
    return "Prostokąt";
}

File: lista3/zadanie1/Rectangle.java
import java.util.*;

class Rectangle extends Quadrangle
{
    public static boolean CheckIfItCouldBe(ClientInput input)
    {
        Set<Double> checkSides = readSidesFromInput(input);

        if(checkSides.size() > 2)
            return false;
        if(input.NextInput() != 90.)
            return false;
        return true;
    }

    public Rectangle(ClientInput input)
    {
        super(input);
    }

    public double Area()
    {
        Iterator<Double> iterator = sides.iterator();
        return iterator.next() * iterator.next();
    }

    public double Perimeter()
    {
        Iterator<Double> iterator = sides.iterator();
        return 2 * iterator.next()  + 2 * iterator.next();
    }

    public String NameYourself()
    {
        return "Prostokąt";
    }

};
